docs/guides/git/index.md

```diff
@@ -0,0 +1,473 @@
+
+```
+    git config --global user.name "Your Name"
+    git config --global user.email yourmail@domain.tld
+```
+
+Unless you are the repository maintainer, it is wise to set your Git to not push changes in file permissions:
+
+```
+    git config --global core.filemode false
+```
+
+It's recommended to verify that the your Git installation is not performing any transformation between LFs and CRLFs. Moodle use **only LFs** and you should configure your editor/IDE the same way. Note that having any "magic" enabled is known to cause [problems with unit tests](https://docs.moodle.org/dev/Common_unit_test_problems#The_test_file_.22evolution.test.22_should_not_contain_section_named_.22.5Blots_of_content.5D.22) execution. We recommend you to set:
+
+```
+    git config --global core.autocrlf false
+```
+
+## Setting-up your public repository
+
+1. Go to [Github](http://github.com/) and create an account.
+
+2. Go to the official [Moodle Github repository](http://github.com/moodle/moodle) and click on the Fork button. You now have your own Github Moodle repository.
+
+3. Set up your SSH public key so you can push to your Github Moodle repository from your local Moodle repository. Over at Github, navigate to your Settings page and look for a section called **SSH and GPG keys**. Configuring your SSH key can be a little tricky, but following [Github's SSH setup steps](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent) will ensure the hardest part of this setup is done correctly.
+
+4. Optionally, you can set up [Github Actions](/general/development/tools/gha) to automatically test your code. This provides the advantage of catching issues without having to set up a testing environment locally and provides helpful feedback before you submit your code for review.
+
+## Setting-up your local repository
+
+Create a local clone of your Github repository by using one of these commands in your terminal:
+
+```
+    git clone git://github.com/YOUR_GITHUB_USERNAME/moodle.git YOUR_LOCALDIR
+```
+
+```
+    git clone git@github.com:YOUR_GITHUB_USERNAME/moodle.git YOUR_LOCALDIR
+```
+
+These commands do several jobs for you:
+
+- Creates a new folder
+- Initialises a new local Git repository
+- Sets your Github repository as the remote repository called 'origin'
+- Makes a local checkout of the branch 'master'
+
+:::note
+In the first example, the URL used in the command is `git://github.com`. This gives read-only access to the your Github repository, and the `git push origin` command that appears later in this document will not work.
+
+If you want to be able to update your **origin** repository, use the URL that starts with `git@github.com` (the second of the two `git clone` commands given above). This will give you read and write access to your Github repository at github.com.
```

Comment:
```suggestion
If you want to be able to update your `origin` repository, use the URL that starts with `git@github.com` (the second of the two `git clone` commands given above). This will give you read and write access to your Github repository at github.com.
```

Reply:
I have removed this reference as we are offering only one suggestion to clone now. Just use the git@github.com...